### PR TITLE
fix(mayactl): invalid status error when length of stats is not same

### PR DESCRIPTION
**Pull request by ashishranjan738**
Signed-off-by: Ashish Ranjan <ashishranjan738@gmail.com>

**What this PR does / why we need it**:
-> Removed the length check as the length of metrics can change on each request.
-> Assumed the metrics family to be 0 if not present in the stats.

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

**Special notes for your reviewer**:

### CODE REVIEW

**Comment by kmova** [comment]
@ashishranjan738, what is the cause for change in metrics length?

**Comment by ashishranjan738** [comment]
@kmova seems like some field like ```openebs_connection_retry``` gets added to some responses and this changes its length. This PR fixes that by picking only the required parameters from the response.

File: cmd/mayactl/app/command/volume_stats_test.go [hunk 1]
```diff
@@ -73,6 +74,19 @@ func TestRunVolumeStats(t *testing.T) {
...
+				T:            t,
```
[the remark this answers is not in the document]
Comment (ashishranjan738) [reply cmd/mayactl/app/command/volume_stats_test.go]: It's not redundant metrics @utkarshmani1997 .This one contains the response having empty metrics family.

File: cmd/mayactl/app/command/volume_stats_test.go [hunk 2]
```diff
@@ -130,37 +144,31 @@ func TestRunVolumeStats(t *testing.T) {
...
-				{}, {},
-			},
-			stats2: v1alpha1.VolumeMetricsList{
-				{}, {},
-			},
-			Output: v1alpha1.StatsJSON{},
-			err:    nil,
+			stats1: map[string]v1alpha1.MetricsFamily{},
+			stats2: map[string]v1alpha1.MetricsFamily{},
+			Output: v1alpha1.StatsJSON{IQN: "", Volume: "", Portal: "", Size: "0.000000", CASType: "", ReadIOPS: 0, WriteIOPS: 0, ReadThroughput: 0, WriteThroughput: 0, ReadLatency: 0, WriteLatency: 0, AvgReadBlockSize: 0, AvgWriteBlockSize: 0, SectorSize: 0, ActualUsed: 0, LogicalSize: 0},
```
[the remark this answers is not in the document]
Comment (ashishranjan738) [reply cmd/mayactl/app/command/volume_stats_test.go]: addressed

File: cmd/mayactl/app/command/volume_stats_test.go [hunk 3]
```diff
@@ -130,37 +144,31 @@ func TestRunVolumeStats(t *testing.T) {
...
 			},
-			stats2: v1alpha1.VolumeMetricsList{
-				{},
+			stats2: map[string]v1alpha1.MetricsFamily{
+				"test1": {},
+				"test2": {},
 			},
-			Output: v1alpha1.StatsJSON{},
-			err:    errors.New("Invalid Response"),
+			Output: v1alpha1.StatsJSON{IQN: "", Volume: "", Portal: "", Size: "0.000000", CASType: "", ReadIOPS: 0, WriteIOPS: 0, ReadThroughput: 0, WriteThroughput: 0, ReadLatency: 0, WriteLatency: 0, AvgReadBlockSize: 0, AvgWriteBlockSize: 0, SectorSize: 0, ActualUsed: 0, LogicalSize: 0},
```
[the remark this answers is not in the document]
Comment (ashishranjan738) [reply cmd/mayactl/app/command/volume_stats_test.go]: addressed

File: cmd/mayactl/app/command/volume_stats_test.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (utkarshmani1997) [reply cmd/mayactl/app/command/volume_stats_test.go]: i mean, we should only keep metrics with prefix `openebs_` and ignore the others such as `go_*`

[the remark this answers is not in the document]
Comment (ashishranjan738) [reply cmd/mayactl/app/command/volume_stats_test.go]: I believe we should keep the original response structure that is coming  from the endpoint.

**Pull request merged**